lib/posix/pthread.c

```diff
@@ -268,15 +265,14 @@ int pthread_cancel(pthread_t pthread)
 	k_spin_unlock(&thread->cancel_lock, cancel_key);
 
 	if (cancel_state == PTHREAD_CANCEL_ENABLE) {
-		pthread_mutex_lock(&thread->state_lock);
+		state_key = k_spin_lock(&thread->state_lock);
 		if (thread->state == PTHREAD_DETACHED) {
 			thread->state = PTHREAD_TERMINATED;
 		} else {
 			thread->retval = PTHREAD_CANCELED;
 			thread->state = PTHREAD_EXITED;
-			pthread_cond_broadcast(&thread->state_cond);
```

Comment:
What was this broadcast used for? And, if necessary, how is it being replaced?

Reply:
This was removed by Andy. I assumed that `k_thread_join()` would notify other waiting threads. I'll double check.